test/integration/particle_emitter.cc

```diff
@@ -0,0 +1,111 @@
+#include "ignition/gazebo/test_config.hh"
+
+#include "helpers/Relay.hh"
+#include "helpers/UniqueTestDirectoryEnv.hh"
+
+using namespace ignition;
+using namespace gazebo;
+
+class ParticleEmitterTest : public ::testing::Test
+{
+  // Documentation inherited
+  protected: void SetUp() override
+  {
+    ignition::common::Console::SetVerbosity(4);
+    setenv("IGN_GAZEBO_SYSTEM_PLUGIN_PATH",
+           (std::string(PROJECT_BINARY_PATH) + "/lib").c_str(), 1);
+  }
+  public: void LoadWorld(const std::string &_path, bool _useLevels = false)
+  {
+    this->serverConfig.SetResourceCache(test::UniqueTestDirectoryEnv::Path());
+    this->serverConfig.SetSdfFile(
+        common::joinPaths(PROJECT_SOURCE_PATH, _path));
+    this->serverConfig.SetUseLevels(_useLevels);
+
+    this->server = std::make_unique<Server>(this->serverConfig);
+    EXPECT_FALSE(this->server->Running());
+    EXPECT_FALSE(*this->server->Running(0));
+    using namespace std::chrono_literals;
+    this->server->SetUpdatePeriod(1ns);
+  }
+
+  public: ServerConfig serverConfig;
+  public: std::unique_ptr<Server> server;
+};
+
+/////////////////////////////////////////////////
+// Load an SDF with a particle emitter and verify its properties.
+TEST_F(ParticleEmitterTest, SDFLoad)
+{
+  const ignition::math::Pose3d expectedPose(0, 0, 0, 0, 0, 0);
+  bool updateChecked{false};
+
+  this->LoadWorld("test/worlds/particle_emitter.sdf");
+
+  // Create a system that checks a particle emitter.
+  test::Relay testSystem;
+  testSystem.OnPostUpdate([&](const gazebo::UpdateInfo &,
+                              const gazebo::EntityComponentManager &_ecm)
+      {
+        _ecm.Each<components::ParticleEmitter,
+                  components::Name,
+                  components::Pose>(
+            [&](const ignition::gazebo::Entity &/*_entity*/,
+                const components::ParticleEmitter *_emitter,
+                const components::Name *_name,
+                const components::Pose *_pose) -> bool
+            {
+              updateChecked = true;
+
+              EXPECT_EQ("particle_emitter_smoke_emitter", _name->Data());
+              EXPECT_EQ(expectedPose, _pose->Data());
+              EXPECT_TRUE(_emitter->Data().emitting());
+              EXPECT_DOUBLE_EQ(2, _emitter->Data().lifetime());
+              EXPECT_DOUBLE_EQ(10, _emitter->Data().min_velocity());
+              EXPECT_DOUBLE_EQ(20, _emitter->Data().max_velocity());
+              EXPECT_DOUBLE_EQ(10, _emitter->Data().scale_rate());
+
+              return true;
+            });
+      });
+
+  this->server->AddSystem(testSystem.systemPtr);
+  this->server->Run(true, 1, false);
+
+  EXPECT_TRUE(updateChecked);
+}
+
+/////////////////////////////////////////////////
+/// Main
+int main(int _argc, char **_argv)
+{
+  ::testing::InitGoogleTest(&_argc, _argv);
+  ::testing::AddGlobalTestEnvironment(
+      new test::UniqueTestDirectoryEnv("particle_emitter_test_cache"));
```

Comment:
Is this needed for this test? Based on what I read in [`UniqueTestDirectory.hh`](https://github.com/ignitionrobotics/ign-gazebo/blob/10988c99f847f57b461b65372c54baa16a906507/test/helpers/UniqueTestDirectoryEnv.hh#L32-L53), I don't believe we need this for this test, but let me know if there's something I'm not catching here.

Reply:
I removed it and the test seems to pass fine. ec532c9e9ae83b6e5e43cfb466db18f90f1e921a